cmd/abapEnvironmentCreateSystem.go

```diff
@@ -129,15 +141,53 @@ func generateManifestYAML(config *abapEnvironmentCreateSystemOptions) ([]byte, e
 	return manifestYAML, nil
 }
 
+func checkManifestParameters(config *abapEnvironmentCreateSystemOptions) (err error) {
+
+	//Only checks for correct parameter specification if CF Service "abap-oem" is selected
+	if config.CfService == "abap-oem" {
+
+		if config.AbapSystemParentSaasAppname == "" && config.AbapSystemParentServiceLabel == "" {
+			return errors.New("Both parameters AbapSystemParentServiceLabel and AbapSystemParentSaasAppname seem to be empty. Please specify either AbapSystemParentServiceLabel or AbapSystemParentSaasAppname depending on who created the oem-instance in the step configuration. For more information please refer to the step documentation")
+		}
+
+		if config.AbapSystemParentSaasAppname != "" {
+			const parentSaasAppnameSyntaxCheck = `[a-zA-Z0-9\-\_]+`
+			addonProductVersionMatch, _ := regexp.MatchString(parentSaasAppnameSyntaxCheck, config.AbapSystemParentSaasAppname)
+			if !addonProductVersionMatch {
+				return errors.New("It seems like you have incorrectly specified the AbapSystemParentSaasAppname step parameter. Please check that the parameters follows the respective syntax to specify the AbapSystemParentSaasAppname. For more information please refer to the step documentation")
```

Comment:
I would propose to be more specific in the error message so the consumer knows what led to the failing check. E.g. the AbapSystemParentSaasAppname contains invalid characters...

Reply:
👍